src/components/Header/Header.jsx

```diff
@@ -1,18 +1,101 @@
-import { HeaderContainer, ElementContainer } from './Header.styles';
+import {
+	HeaderContainer,
+	ElementContainer,
+	LogoLinkContainer,
+	Name,
+} from './Header.styles';
+
+import { useState, useEffect, Fragment, useContext } from 'react';
+import { Outlet, useNavigate, useLocation } from 'react-router-dom';
 
 import Logo from './components/Logo/Logo';
 import Button, { BUTTON_TYPES_CLASSES } from '../../common/Button/Button';
 
+import { UserContext } from '../../helpers/context/user.context';
+
 const Header = () => {
+	const navigate = useNavigate();
+	const userToken = localStorage.getItem('userToken');
+
+	const [onAuthPages, setOnAuthPages] = useState(false);
+
+	const { user, setUser, setUserToken } = useContext(UserContext);
+
+	const currentLocation = useLocation(); //Where am i?
+	// useCallback(() => {
+	// 	setLocation(currentLocation);
+	// }, [currentLocation]);
+
+	useEffect(() => {
+		if (
+			currentLocation.pathname.includes('login') ||
+			currentLocation.pathname.includes('registration')
+		) {
+			setOnAuthPages(true);
+		} else {
+			setOnAuthPages(false);
+		}
+	}, [currentLocation]);
+
+	useEffect(
+		() => {
+			if (userToken) {
+				navigate('/courses');
+			} else {
+				navigate('/login');
+			}
+		},
+		// currently only on mount
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+		[]
```

Comment:
Yes, and then it makes sense to add the userToken itself to the array
And not bind to the physical token in the localstorage, but rely only on the context.
Then in the logInOutButtonHandler, you don't need to call navigate, useEffect will do it.
However, you still need to check for its presence in the store.
I'm going to change the logic a little bit to make it as close to the state as possible.
But if you leave the navigate in the array, then it will call useEffect with its changes and then, if the localStateToken exists, it will be impossible to get to the login page.

Perhaps I misunderstood the task in the task.

Because then I need to implement two opposite tasks 
1) Even in the presence of a token, you can come to the /login page
Otherwise, I have nowhere to get the user object
2) If token is in the localStorage app automatically navigates to /courses route.